### PR TITLE
Bug 1769716: Remove storage classes unrelated to the storage resource

### PR DESCRIPTION
**Pull request by bipuladh**
- Filters Noobaa provisioned storage classes out of PVC creation SC selection dropdown.
- Filters SCs not being provisioned by Noobaa out of OBC creation.
Screenshot from 2019-11-08 12-49-10

### CODE REVIEW

File: frontend/packages/noobaa-storage-plugin/src/components/object-bucket-claim-page/create-obc.tsx [hunk 1]
```diff
@@ -21,6 +21,9 @@ import { ActionGroup, Button } from '@patternfly/react-core';
 import { getName } from '@console/shared';
 import { commonReducer, defaultState } from '../object-bucket-page/state';
 
+// Accept only noobaa provisioned
+const filterStorageClasses = (sc) => sc.provisioner === 'openshift-storage.noobaa.io/obc';
```
Comment (cloudbehl) [inline frontend/packages/noobaa-storage-plugin/src/components/object-bucket-claim-page/create-obc.tsx]: create a const for `noobaa.io/obc`
In generic utils of noobaa right the fuction to filter and don't consider `openshift-storage` prefix like we did for ceph.
```suggestion
const filterStorageClasses = (sc) => sc.provisioner === 'openshift-storage.noobaa.io/obc';
```

Comment (bipuladh) [reply frontend/packages/noobaa-storage-plugin/src/components/object-bucket-claim-page/create-obc.tsx]: okay

Comment (bipuladh) [reply frontend/packages/noobaa-storage-plugin/src/components/object-bucket-claim-page/create-obc.tsx]: I moved it to utils. But does it make sense to make a const while it's only being used in one place?

File: frontend/packages/noobaa-storage-plugin/src/utils.ts [hunk 2]
```diff
@@ -26,3 +26,5 @@ export const getResiliencyProgress = (response: PrometheusResponse): number => {
   const progress = getGaugeValue(response);
   return Number(Number(progress).toFixed(1));
 };
+
+export const filterStorageClasses = (sc) => sc.provisioner.includes('noobaa.io/obc');
```
Comment (rawagner) [inline frontend/packages/noobaa-storage-plugin/src/utils.ts]: can you add type for `sc` ?

File: frontend/public/components/storage/create-pvc.tsx [hunk 3]
```diff
@@ -197,6 +197,9 @@ export const CreatePVCForm: React.FC<CreatePVCFormProps> = (props) => {
     setAccessMode(event.currentTarget.value);
   };
 
+  //Accept anything except noobaa provisioned
+  const filterStorageClasses = (sc) => !sc.provisioner.includes('noobaa.io/obc');
```
Comment (rawagner) [inline frontend/public/components/storage/create-pvc.tsx]: I dont think we should have noobaa specific stuff in core console. Shouldn't this be done via some extension point ?

File: frontend/packages/noobaa-storage-plugin/src/utils.ts (hunk 2 above)
Comment (bipuladh) [reply frontend/packages/noobaa-storage-plugin/src/utils.ts]: sc has a weird type it extends upon K8sResource.
https://github.com/openshift/console/blob/master/frontend/public/components/utils/storage-class-dropdown.tsx#L47-L62

File: frontend/public/components/storage/create-pvc.tsx (hunk 3 above)
Comment (rawagner) [reply frontend/public/components/storage/create-pvc.tsx]: actually there's already a ceph specific stuff which just got moved to `storage-utils` see https://github.com/openshift/console/pull/3312 . You could use a similar approach

File: frontend/packages/noobaa-storage-plugin/src/utils.ts (hunk 2 above)
Comment (rawagner) [reply frontend/packages/noobaa-storage-plugin/src/utils.ts]: not needed anymore as you are using the one from `console-shared`

File: frontend/packages/console-shared/src/utils/storage-utils.ts [hunk 4]
```diff
@@ -12,3 +12,6 @@ export const isCephProvisioner = (scProvisioner: string): boolean => {
     _.endsWith(scProvisioner, provisioner),
   );
 };
+
+// To check that the storage class isn't noobaa based
+export const filterStorageClasses = (sc) => !sc.provisioner.includes('noobaa.io/obc');
```
Comment (rawagner) [inline frontend/packages/console-shared/src/utils/storage-utils.ts]: I think you can use `StorageClass` type from `@console/internal/components/storage-class-form`

```suggestion
export const filterStorageClasses = (sc: StorageClass) => !sc.provisioner.includes('noobaa.io/obc');
```

File: frontend/packages/console-shared/src/utils/storage-utils.ts [hunk 5]
```diff
@@ -12,3 +13,9 @@ export const isCephProvisioner = (scProvisioner: string): boolean => {
     _.endsWith(scProvisioner, provisioner),
   );
 };
+
+// To check that the storage class isn't noobaa based
+export const filterScOnProvisioner = (provisioner: string = '', show: boolean = true) =>
```
Comment (rawagner) [inline frontend/packages/console-shared/src/utils/storage-utils.ts]: we could simplify this to 
```suggestion
export const filterScOnProvisioner = (sc: StorageClass, provisioner: string): boolean => sc.provisioner.includes(provisioner);
```
and use `useCallback` later

also the comment above does not apply anymore as you can specify any storage class, not just noobaa

File: frontend/packages/noobaa-storage-plugin/src/components/object-bucket-claim-page/create-obc.tsx [hunk 6]
```diff
@@ -64,6 +64,8 @@ export const CreateOBCPage: React.FC<CreateOBCPageProps> = (props) => {
       });
   };
 
+  const showOnlyObcScs = React.useMemo(() => filterScOnProvisioner('noobaa.io/obc'), []);
```
Comment (rawagner) [inline frontend/packages/noobaa-storage-plugin/src/components/object-bucket-claim-page/create-obc.tsx]:
```suggestion
  const showOnlyObcScs = React.useCallback((sc: StorageClass) => filterScOnProvisioner(sc, 'noobaa.io/obc'), []);
```

File: frontend/public/components/storage/create-pvc.tsx [hunk 7]
```diff
@@ -196,6 +196,8 @@ export const CreatePVCForm: React.FC<CreatePVCFormProps> = (props) => {
     setAccessMode(event.currentTarget.value);
   };
 
+  const onlyPvcSCs = React.useMemo(() => filterScOnProvisioner('noobaa.io/obc', false), []);
```
Comment (rawagner) [inline frontend/public/components/storage/create-pvc.tsx]:
```suggestion
  const onlyPvcSCs = React.useCallback((sc: StorageClass) => !filterScOnProvisioner(sc, 'noobaa.io/obc'), []);
```

**Pull request merged**